app/javascript/gobierto_data/webapp/components/Dataset.vue

```diff
@@ -491,11 +627,28 @@ export default {
         const { sql: stringError } = sqlError
         this.queryError = stringError
         this.isQueryRunning = false;
+        this.enabledQuerySavedButton = false
       }
     },
     async storeCurrentVisualization(config, opts) {
 
-      const { name, privacy } = opts;
+      const { name, privacy, vizID, user, queryViz } = opts;
+
+      const userId = Number(getUserId());
+
+      // Get the id if the query matches with a stored query
+      const { id } =
+        this.privateQueries.find(
+          ({ attributes: { sql } }) => sql === this.currentQuery
+        ) || {};
+
+      let currentQueryViz;
+
+      if (user !== userId) {
```

Comment:
This inequality means you're trying to save a query from another user?

Reply:
Yes, in this case, the user is trying to save a Viz from another user